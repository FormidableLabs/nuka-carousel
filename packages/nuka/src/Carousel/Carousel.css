.nuka-hidden {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border-width: 0;
}

.nuka-container {
  position: relative;
}

.nuka-slide-container {
  position: relative;
}

.nuka-overflow {
  overflow: scroll;
  scroll-behavior: smooth;
  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */
}

.nuka-overflow::-webkit-scrollbar {
  display: none;
}

.nuka-wrapper {
  display: flex;
}

.nuka-wrapper--scroll-snapping {
  scroll-snap-type: x mandatory;
}

.nuka-nav-button {
  position: absolute;
  top: calc(50% - 2rem);
  margin: 1rem;
  display: none;
  height: 2rem;
  width: 2rem;
  cursor: pointer;
  background-color: rgba(146, 148, 151, 0.5);
  color: white;
  border-radius: 9999px;
  font-size: 1rem;
  user-select: none;

  &.nuka-nav-button-prev {
    left: 0;
    margin-right: 1rem;
  }
  &.nuka-nav-button-next {
    right: 0;
    margin-left: 1rem;
  }
  &:hover {
    background-color: rgba(146, 148, 151, 0.65);
  }
}

.nuka-nav-button-enabled {
  display: block;
}

.nuka-container-auto-hide {
  .nuka-nav-button {
    display: none;
  }

  &:hover .nuka-nav-button {
    display: block;
  }
}

.nuka-page-container {
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  gap: 0.5rem;
  justify-content: center;
  align-items: center;
}

.nuka-page-indicator {
  width: 0.75rem;
  height: 0.75rem;
  cursor: pointer;
  border-radius: 9999px;
  border-style: none;
  background-color: rgba(146, 148, 151, 0.65);

  &.nuka-page-indicator-active,
  &.nuka-page-indicator-active:hover {
    background-color: rgb(229 231 235 / 1);
  }

  &:hover {
    background-color: rgb(229 231 235 / 1);
  }
}
