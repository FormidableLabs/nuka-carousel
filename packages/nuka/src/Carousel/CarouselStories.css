.slide__with-gap {
  gap: 8px;
}
