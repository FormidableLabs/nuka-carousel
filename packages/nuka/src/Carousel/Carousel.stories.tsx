import type { Meta, StoryObj } from '@storybook/react';
import { Carousel, CarouselProps, SlideHandle } from './Carousel';
import { useRef } from 'react';
import { ExampleSlide } from './ExampleSlide';
import './CarouselStories.css';

const StorybookComponent = (props: CarouselProps) => {
  const ref = useRef<SlideHandle>(null);
  return (
    <div>
      <Carousel ref={ref} {...props} />
      <button
        onClick={() => {
          if (ref.current) ref.current.previousSlide();
        }}
      >
        previous
      </button>
      <button
        onClick={() => {
          if (ref.current) ref.current.nextSlide();
        }}
      >
        next
      </button>
    </div>
  );
};

const meta: Meta<typeof Carousel> = {
  title: 'components/Carousel',
  component: StorybookComponent,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof Carousel>;

export const Default: Story = {
  args: {
    children: (
      <>
        {[...Array(6)].map((_, index) => (
          <ExampleSlide key={index} index={index} />
        ))}
      </>
    ),
  },
};

export const Slide: Story = {
  args: {
    scrollDistance: 'slide',
    wrapperClassName: 'slide__with-gap',
    children: (
      <>
        {[...Array(10)].map((_, index) => (
          <ExampleSlide key={index} index={index} />
        ))}
      </>
    ),
  },
};
