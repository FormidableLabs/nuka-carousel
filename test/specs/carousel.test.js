import Carousel from '../../src';

describe('<Carousel />', () => {
  describe('Rendering and Mounting', () => {
    it('should correctly mount with children.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const children = wrapper.find('p');
      expect(children).toHaveLength(3);
    });

    it('should render a div with the class `slider`.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const sliderDiv = wrapper.find('div.slider');
      expect(sliderDiv).toHaveLength(1);
    });

    it('should render a div with the class `slider-frame`.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const sliderFrameDiv = wrapper.find('div.slider-frame');
      expect(sliderFrameDiv).toHaveLength(1);
    });

    it('should render an ul with the class `slider-list`.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const sliderListUl = wrapper.find('ul.slider-list');
      expect(sliderListUl).toHaveLength(1);
    });

    it('should render children with the `slider-slide` class.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const children = wrapper.find('.slider-slide');
      expect(children).toHaveLength(3);
    });

    it('should render controls by default.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const decorator1 = wrapper.find('.slider-control-centerleft');
      const decorator2 = wrapper.find('.slider-control-centerright');
      const decorator3 = wrapper.find('.slider-control-bottomcenter');
      expect(decorator1).toHaveLength(1);
      expect(decorator2).toHaveLength(1);
      expect(decorator3).toHaveLength(1);
    });
  });

  describe('Props', () => {
    it('should render with the class `slider` when no props are supplied.', () => {
      const wrapper = mount(
        <Carousel>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const slider = wrapper.find('div.slider');
      expect(slider).toHaveLength(1);
    });

    it('should render with the class `test` with className supplied.', () => {
      const wrapper = mount(
        <Carousel className="test">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const slider = wrapper.find('div.test');
      expect(slider).toHaveLength(1);
    });

    it('should merge provided styles with default styles.', () => {
      const wrapper = mount(
        <Carousel style={{ backgroundColor: 'black' }}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const slider = wrapper.find('div.slider');
      expect(slider).toHaveStyle('backgroundColor', 'black');
      expect(slider).toHaveStyle('display', 'block');
    });

    it('should align to 0 when `cellAlign` is `left`.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const slider = wrapper.find('.slider-list');
      expect(slider).toHaveStyle('transform', 'translate3d(0px, 0px, 0)');
    });

    it('should allow navigation to the last slide for center align and > 1 slidesToShow.', () => {
      const wrapper = mount(
        <Carousel cellAlign="center" slidesToShow={3}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
          <p>Slide 4</p>
          <p>Slide 5</p>
          <p>Slide 6</p>
        </Carousel>
      );
      const nextButton = wrapper.find('.slider-control-centerright button');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      expect(wrapper).toHaveState({ currentSlide: 5 });
    });

    it('should allow navigation to the last slide for right align and > 1 slidesToShow.', () => {
      const wrapper = mount(
        <Carousel cellAlign="right" slidesToShow={2}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
          <p>Slide 4</p>
          <p>Slide 5</p>
          <p>Slide 6</p>
        </Carousel>
      );
      const nextButton = wrapper.find('.slider-control-centerright button');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      nextButton.simulate('click');
      expect(wrapper).toHaveState({ currentSlide: 5 });
    });

    it('should adjust the slide index when children count change.', () => {
      const wrapper = mount(
        <Carousel slideIndex={2}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState('currentSlide', 2);
      wrapper.setProps({
        children: [<p key={1}>Slide 1</p>, <p key={2}>Slide 2</p>]
      });
      expect(wrapper).toHaveState('currentSlide', 1);
    });

    it('should disable controls when children are updated to have a length of 0.', () => {
      const wrapper = mount(
        <Carousel slideIndex={2}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState('currentSlide', 2);
      wrapper.setProps({ children: [] });
      expect(wrapper).toHaveState('currentSlide', 0);

      const previousButton = wrapper.find('.slider-control-centerleft button');
      const nextButton = wrapper.find('.slider-control-centerright button');
      expect(previousButton).toHaveProp('disabled', true);
      expect(nextButton).toHaveProp('disabled', true);
    });

    it('should set slideHeight to max value when `heightMode` is `max`', () => {
      jest
        .spyOn(Carousel.prototype, 'getChildNodes')
        .mockReturnValue([
          { offsetHeight: 100, style: {} },
          { offsetHeight: 200, style: {} },
          { offsetHeight: 300, style: {} }
        ]);
      const wrapper = mount(
        <Carousel heightMode="max">
          <div style={{ height: '100px' }}>Slide 1</div>
          <div style={{ height: '200px' }}>Slide 1</div>
          <div style={{ height: '300px' }}>Slide 1</div>
        </Carousel>
      );
      Carousel.prototype.getChildNodes.mockRestore();
      expect(wrapper).toHaveState({ slideHeight: 300 });
    });

    it("should set slideHeight to first slide's height when `heightMode` is `first`", () => {
      jest
        .spyOn(Carousel.prototype, 'getChildNodes')
        .mockReturnValue([
          { offsetHeight: 100, style: {} },
          { offsetHeight: 200, style: {} },
          { offsetHeight: 300, style: {} }
        ]);
      const wrapper = mount(
        <Carousel heightMode="first">
          <div style={{ height: '100px' }}>Slide 1</div>
          <div style={{ height: '200px' }}>Slide 1</div>
          <div style={{ height: '300px' }}>Slide 1</div>
        </Carousel>
      );
      Carousel.prototype.getChildNodes.mockRestore();
      expect(wrapper).toHaveState({ slideHeight: 100 });
    });

    it('should set height to current slide height when `heightMode` is `current`', () => {
      jest
        .spyOn(Carousel.prototype, 'getChildNodes')
        .mockReturnValue([
          { offsetHeight: 100, style: {} },
          { offsetHeight: 200, style: {} },
          { offsetHeight: 300, style: {} }
        ]);
      const wrapper = mount(
        <Carousel heightMode="current" slideIndex={1}>
          <div style={{ height: '100px' }}>Slide 1</div>
          <div style={{ height: '200px' }}>Slide 1</div>
          <div style={{ height: '300px' }}>Slide 1</div>
        </Carousel>
      );
      Carousel.prototype.getChildNodes.mockRestore();
      expect(wrapper).toHaveState({ slideHeight: 200 });
    });
  });

  describe('methods', () => {
    it('should advance if nextSlide() is called.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState({ currentSlide: 0 });
      wrapper.instance().nextSlide();
      expect(wrapper).toHaveState({ currentSlide: 1 });
    });

    it('should not advance past the last slide.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      wrapper.instance().nextSlide();
      wrapper.instance().nextSlide();
      expect(wrapper).toHaveState({ currentSlide: 2 });
      wrapper.instance().nextSlide();
      expect(wrapper).toHaveState({ currentSlide: 2 });
    });

    it('should not go back to the previous slide when index is 0.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState({ currentSlide: 0 });
      wrapper.instance().previousSlide();
      expect(wrapper).toHaveState({ currentSlide: 0 });
    });

    it('should go back to the previous slide when `previousSlide` is called.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState({ currentSlide: 0 });
      wrapper.instance().nextSlide();
      wrapper.instance().nextSlide();
      wrapper.instance().previousSlide();
      expect(wrapper).toHaveState({ currentSlide: 1 });
    });

    it('should go to correct slide when `goToSlide` is called.', () => {
      const wrapper = mount(
        <Carousel cellAlign="left">
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toHaveState({ currentSlide: 0 });
      wrapper.instance().goToSlide(2);
      expect(wrapper).toHaveState({ currentSlide: 2 });
    });

    it('should go to the last slide from the first when wrapAround is true`.', () => {
      const wrapper = mount(
        <Carousel wrapAround>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const previousButton = wrapper.find('.slider-control-centerleft button');
      expect(wrapper).toHaveState({ currentSlide: 0 });
      previousButton.simulate('click');
      expect(wrapper).toHaveState({ currentSlide: 2 });
    });

    it('should go to the first slide from the last when wrapAround is true`.', () => {
      const wrapper = mount(
        <Carousel wrapAround>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      wrapper.setState({ currentSlide: 2 });
      const nextButton = wrapper.find('.slider-control-centerright button');
      expect(wrapper).toHaveState({ currentSlide: 2 });
      nextButton.simulate('click');
      expect(wrapper).toHaveState({ currentSlide: 0 });
    });
  });

  describe('Controls', () => {
    it('should render a custom top left control.', () => {
      const wrapper = mount(
        <Carousel renderTopLeftControls={() => <div>Top Left</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Top Left</div>);
    });

    it('should render a custom top center control.', () => {
      const wrapper = mount(
        <Carousel renderTopCenterControls={() => <div>Top Center</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Top Center</div>);
    });

    it('should render a custom top right control.', () => {
      const wrapper = mount(
        <Carousel renderTopRightControls={() => <div>Top Right</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Top Right</div>);
    });

    it('should render a custom center left control.', () => {
      const wrapper = mount(
        <Carousel renderCenterLeftControls={() => <div>Center Left</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Center Left</div>);
    });

    it('should render a custom center center control.', () => {
      const wrapper = mount(
        <Carousel renderCenterCenterControls={() => <div>Center Center</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Center Center</div>);
    });

    it('should render a custom center right control.', () => {
      const wrapper = mount(
        <Carousel renderCenterRightControls={() => <div>Center Right</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Center Right</div>);
    });

    it('should render a custom bottom left control.', () => {
      const wrapper = mount(
        <Carousel renderBottomLeftControls={() => <div>Bottom Left</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Bottom Left</div>);
    });

    it('should render a custom bottom center control.', () => {
      const wrapper = mount(
        <Carousel renderBottomCenterControls={() => <div>Bottom Center</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Bottom Center</div>);
    });

    it('should render a custom bottom right control.', () => {
      const wrapper = mount(
        <Carousel renderBottomRightControls={() => <div>Bottom Right</div>}>
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      expect(wrapper).toContainReact(<div>Bottom Right</div>);
    });

    it('should render controls with control props.', () => {
      const CustomControls = () => <div>Custom Controls</div>;
      const wrapper = mount(
        <Carousel
          wrapAround
          renderCenterCenterControls={props => <CustomControls {...props} />}
        >
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const controls = wrapper.find('CustomControls');
      expect(controls).toHaveProp('currentSlide', 0);
      expect(controls).toHaveProp('slideCount', 3);
      expect(controls).toHaveProp('wrapAround', true);
      expect(controls).toHaveProp('nextSlide');
      expect(controls).toHaveProp('previousSlide');
      expect(controls).toHaveProp('goToSlide');
    });

    it('should call the internal nextSlide func from a control using the nextSlide prop.', () => {
      const wrapper = mount(
        <Carousel
          renderCenterCenterControls={({ nextSlide }) => (
            <button id="custom-next-btn" onClick={nextSlide}>
              Next
            </button>
          )}
        >
          >
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const spy = jest.spyOn(wrapper.instance(), 'nextSlide');
      wrapper.update();
      const button = wrapper.find('#custom-next-btn');
      button.simulate('click');
      expect(spy).toHaveBeenCalled();
    });

    it('should call the internal previousSlide func from a control using the previousSlide prop.', () => {
      const wrapper = mount(
        <Carousel
          slideIndex={1}
          renderCenterCenterControls={({ previousSlide }) => (
            <button id="custom-prev-btn" onClick={previousSlide}>
              Next
            </button>
          )}
        >
          >
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const spy = jest.spyOn(wrapper.instance(), 'previousSlide');
      wrapper.update();
      const button = wrapper.find('#custom-prev-btn');
      button.simulate('click');
      expect(spy).toHaveBeenCalled();
    });

    it('should call the internal goToSlide func from a control using the goToSlide prop.', () => {
      const wrapper = mount(
        <Carousel
          renderCenterCenterControls={({ goToSlide }) => (
            <button id="custom-goto-btn" onClick={() => goToSlide(2)}>
              Go to Slide 3
            </button>
          )}
        >
          >
          <p>Slide 1</p>
          <p>Slide 2</p>
          <p>Slide 3</p>
        </Carousel>
      );
      const spy = jest.spyOn(wrapper.instance(), 'goToSlide');
      wrapper.update();
      const button = wrapper.find('#custom-goto-btn');
      button.simulate('click');
      expect(spy).toHaveBeenCalledWith(2);
    });
  });
});
