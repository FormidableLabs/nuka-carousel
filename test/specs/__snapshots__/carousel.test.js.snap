// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> Rendering and Mounting should correctly mount with children and scrollMode set to \`page\`. 1`] = `
<Carousel
  afterSlide={[Function]}
  autoGenerateStyleTag={true}
  autoplay={false}
  autoplayInterval={3000}
  autoplayReverse={false}
  beforeSlide={[Function]}
  cellAlign="left"
  cellSpacing={10}
  defaultControlsConfig={Object {}}
  disableAnimation={false}
  disableEdgeSwiping={false}
  dragging={true}
  easing="easeCircleOut"
  edgeEasing="easeElasticOut"
  enableKeyboardControls={false}
  frameOverflow="hidden"
  framePadding="0px"
  getControlsContainerStyles={[Function]}
  height="inherit"
  heightMode="max"
  keyCodeConfig={Object {}}
  onDragStart={[Function]}
  onResize={[Function]}
  pauseOnHover={true}
  renderAnnounceSlideMessage={[Function]}
  renderBottomCenterControls={[Function]}
  renderCenterLeftControls={[Function]}
  renderCenterRightControls={[Function]}
  scrollMode="page"
  slideIndex={0}
  slideListMargin={10}
  slideOffset={25}
  slideWidth="300"
  slidesToScroll={1}
  slidesToShow={1}
  speed={500}
  style={Object {}}
  swiping={true}
  transitionMode="scroll"
  vertical={false}
  width="100%"
  withoutControls={false}
  wrapAround={false}
>
  <section
    aria-label="carousel-slider"
    className="slider"
    onBlur={[Function]}
    onFocus={[Function]}
    role="region"
    style={
      Object {
        "MozBoxSizing": "border-box",
        "boxSizing": "border-box",
        "display": "block",
        "height": "inherit",
        "position": "relative",
        "width": "100%",
      }
    }
    tabIndex={0}
  >
    <AnnounceSlide
      message="Slide 1 of 3"
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        style={
          Object {
            "border": 0,
            "clip": "rect(0, 0, 0, 0)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        tabIndex={-1}
      >
        Slide 1 of 3
      </div>
    </AnnounceSlide>
    <div
      className="slider-frame"
      onClickCapture={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "MozBoxSizing": "border-box",
          "WebkitTransform": "translate3d(0, 0, 0)",
          "boxSizing": "border-box",
          "display": "block",
          "height": "100%",
          "margin": "0px",
          "msTransform": "translate(0, 0)",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "touchAction": "pinch-zoom pan-y",
          "transform": "translate3d(0, 0, 0)",
        }
      }
    >
      <Animate
        interpolation={[Function]}
        show={true}
        start={
          Object {
            "tx": -0,
            "ty": 0,
          }
        }
        update={[Function]}
      >
        <NodeGroup
          data={
            Array [
              Object {
                "tx": -0,
                "ty": 0,
              },
            ]
          }
          enter={[Function]}
          interpolation={[Function]}
          keyAccessor={[Function]}
          leave={[Function]}
          start={[Function]}
          update={[Function]}
        >
          <ScrollTransition
            cellAlign="left"
            cellSpacing={10}
            currentSlide={0}
            deltaX={-0}
            deltaY={0}
            dragging={true}
            frameWidth={0}
            hasInteraction={false}
            heightMode="max"
            isWrappingAround={false}
            left={0}
            slideCount={3}
            slideHeight={0}
            slideListMargin={10}
            slideOffset={25}
            slideWidth={300}
            slidesToScroll={0}
            slidesToShow={0}
            top={0}
            vertical={false}
            wrapAround={false}
            zoomScale={0.85}
          >
            <div
              className="slider-list"
              style={
                Object {
                  "MozBoxSizing": "border-box",
                  "WebkitTransform": "translate3d(0px, 0px, 0)",
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "block",
                  "height": 0,
                  "margin": "0px -5px",
                  "msTransform": "translate(0px, 0px)",
                  "padding": 0,
                  "position": "relative",
                  "touchAction": "pinch-zoom pan-y",
                  "transform": "translate3d(0px, 0px, 0)",
                  "transition": "0s",
                  "width": "auto",
                }
              }
            >
              <div
                aria-label="slide 1 of 3"
                className="slider-slide"
                key="0/.$.$.0"
                onClick={[Function]}
                role="group"
                style={
                  Object {
                    "MozBoxSizing": "border-box",
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "height": "auto",
                    "left": 0,
                    "listStyleType": "none",
                    "marginBottom": "auto",
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": "auto",
                    "position": "absolute",
                    "top": 0,
                    "transform": "scale(1)",
                    "transition": "transform .4s linear",
                    "verticalAlign": "top",
                    "width": 300,
                  }
                }
                tabIndex={-1}
              >
                <p
                  aria-hidden="false"
                  key=".$.0"
                  tabIndex={0}
                >
                  Slide 1
                </p>
              </div>
              <div
                aria-label="slide 2 of 3"
                className="slider-slide"
                inert="true"
                key="1/.$.$.1"
                onClick={[Function]}
                role="group"
                style={
                  Object {
                    "MozBoxSizing": "border-box",
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "height": "auto",
                    "left": 310,
                    "listStyleType": "none",
                    "marginBottom": "auto",
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": "auto",
                    "position": "absolute",
                    "top": 0,
                    "transform": "scale(1)",
                    "transition": "transform .4s linear",
                    "verticalAlign": "top",
                    "width": 300,
                  }
                }
                tabIndex={-1}
              >
                <p
                  aria-hidden="true"
                  key=".$.1"
                >
                  Slide 2
                </p>
              </div>
              <div
                aria-label="slide 3 of 3"
                className="slider-slide"
                inert="true"
                key="2/.$.$.2"
                onClick={[Function]}
                role="group"
                style={
                  Object {
                    "MozBoxSizing": "border-box",
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "height": "auto",
                    "left": 620,
                    "listStyleType": "none",
                    "marginBottom": "auto",
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": "auto",
                    "position": "absolute",
                    "top": 0,
                    "transform": "scale(1)",
                    "transition": "transform .4s linear",
                    "verticalAlign": "top",
                    "width": 300,
                  }
                }
                tabIndex={-1}
              >
                <p
                  aria-hidden="true"
                  key=".$.2"
                >
                  Slide 3
                </p>
              </div>
            </div>
          </ScrollTransition>
        </NodeGroup>
      </Animate>
    </div>
    <div
      className="slider-control-centerleft"
      key="CenterLeft"
      style={
        Object {
          "WebkitTransform": "translateY(-50%)",
          "left": 0,
          "msTransform": "translateY(-50%)",
          "position": "absolute",
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    >
      <PreviousButton
        cellAlign="left"
        cellSpacing={10}
        currentSlide={0}
        defaultControlsConfig={Object {}}
        frameWidth={0}
        goToSlide={[Function]}
        left={0}
        nextSlide={[Function]}
        previousSlide={[Function]}
        scrollMode="page"
        slideCount={3}
        slideWidth={300}
        slidesToScroll={0}
        slidesToShow={0}
        top={0}
        vertical={false}
        wrapAround={false}
      >
        <button
          aria-label="previous"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(0,0,0,0.4)",
              "border": 0,
              "color": "white",
              "cursor": "not-allowed",
              "opacity": 0.3,
              "padding": 10,
              "textTransform": "uppercase",
            }
          }
          type="button"
        >
          Prev
        </button>
      </PreviousButton>
    </div>
    <div
      className="slider-control-centerright"
      key="CenterRight"
      style={
        Object {
          "WebkitTransform": "translateY(-50%)",
          "msTransform": "translateY(-50%)",
          "position": "absolute",
          "right": 0,
          "top": "50%",
          "transform": "translateY(-50%)",
        }
      }
    >
      <NextButton
        cellAlign="left"
        cellSpacing={10}
        currentSlide={0}
        defaultControlsConfig={Object {}}
        frameWidth={0}
        goToSlide={[Function]}
        left={0}
        nextSlide={[Function]}
        previousSlide={[Function]}
        scrollMode="page"
        slideCount={3}
        slideWidth={300}
        slidesToScroll={0}
        slidesToShow={0}
        top={0}
        vertical={false}
        wrapAround={false}
      >
        <button
          aria-label="next"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "background": "rgba(0,0,0,0.4)",
              "border": 0,
              "color": "white",
              "cursor": "pointer",
              "opacity": false,
              "padding": 10,
              "textTransform": "uppercase",
            }
          }
          type="button"
        >
          Next
        </button>
      </NextButton>
    </div>
    <div
      className="slider-control-bottomcenter"
      key="BottomCenter"
      style={
        Object {
          "WebkitTransform": "translateX(-50%)",
          "bottom": 0,
          "left": "50%",
          "msTransform": "translateX(-50%)",
          "position": "absolute",
          "transform": "translateX(-50%)",
        }
      }
    >
      <PagingDots
        cellAlign="left"
        cellSpacing={10}
        currentSlide={0}
        defaultControlsConfig={Object {}}
        frameWidth={0}
        goToSlide={[Function]}
        left={0}
        nextSlide={[Function]}
        previousSlide={[Function]}
        scrollMode="page"
        slideCount={3}
        slideWidth={300}
        slidesToScroll={0}
        slidesToShow={0}
        top={0}
        vertical={false}
        wrapAround={false}
      >
        <ul
          style={
            Object {
              "display": "flex",
              "listStyleType": "none",
              "margin": 0,
              "padding": 0,
              "position": "relative",
              "top": -10,
            }
          }
        >
          <li
            className="paging-item active"
            key="0"
          >
            <button
              aria-label="slide 1 bullet"
              aria-selected={true}
              onClick={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fill": "black",
                  "opacity": 1,
                }
              }
              type="button"
            >
              <svg
                aria-hidden="true"
                className="paging-dot"
                focusable="false"
                height="6"
                width="6"
              >
                <circle
                  cx="3"
                  cy="3"
                  r="3"
                />
              </svg>
            </button>
          </li>
          <li
            className="paging-item"
            key="1"
          >
            <button
              aria-label="slide 2 bullet"
              aria-selected={false}
              onClick={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fill": "black",
                  "opacity": 0.5,
                }
              }
              type="button"
            >
              <svg
                aria-hidden="true"
                className="paging-dot"
                focusable="false"
                height="6"
                width="6"
              >
                <circle
                  cx="3"
                  cy="3"
                  r="3"
                />
              </svg>
            </button>
          </li>
          <li
            className="paging-item"
            key="2"
          >
            <button
              aria-label="slide 3 bullet"
              aria-selected={false}
              onClick={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fill": "black",
                  "opacity": 0.5,
                }
              }
              type="button"
            >
              <svg
                aria-hidden="true"
                className="paging-dot"
                focusable="false"
                height="6"
                width="6"
              >
                <circle
                  cx="3"
                  cy="3"
                  r="3"
                />
              </svg>
            </button>
          </li>
          <li
            className="paging-item"
            key="3"
          >
            <button
              aria-label="slide 4 bullet"
              aria-selected={false}
              onClick={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fill": "black",
                  "opacity": 0.5,
                }
              }
              type="button"
            >
              <svg
                aria-hidden="true"
                className="paging-dot"
                focusable="false"
                height="6"
                width="6"
              >
                <circle
                  cx="3"
                  cy="3"
                  r="3"
                />
              </svg>
            </button>
          </li>
        </ul>
      </PagingDots>
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".slider-slide > img { width: 100%; display: block; }
          .slider-slide > img:focus { margin: auto; }",
        }
      }
      type="text/css"
    />
  </section>
</Carousel>
`;
